core/domain/auth_validators.py

```diff
@@ -31,18 +31,20 @@ class UserIdByFirebaseAuthIdModelValidator(
     """Class for validating UserIdByFirebaseAuthIdModels."""
 
     @classmethod
-    def _get_model_id_regex(cls, unused_item):
-        """Returns a regex for model id.
-
-        This method can be overridden by subclasses, if needed.
+    def _validate_model_id(cls, item):
+        """Checks whether the id of model matches the regex specified for
+        the model.
 
         Args:
-            unused_item: datastore_services.Model. Entity to validate.
-
-        Returns:
-            str. A regex pattern to be followed by the model id.
+            item: datastore_services.Model. Entity to validate.
         """
-        return '^[\x00-\x7F]{1,255}$'
+        # Firebase only constrains IDs to be between 1 and 128 characters:
```

Comment:
Specify charset too?

Reply:
Hmm this is genuinely the only constraint, so I think charset is implied to not matter right?

Reply:
Can you check what the typical charset from firebase is, in your manual testing? Let's use that.

Reply:
Resolved offline, found alphanumeric works. Done.